nebula-query-and-search/main/classes/AggregateQuery.cls

```diff
@@ -79,19 +80,48 @@ global class AggregateQuery extends SOQL {
     return this.havingAggregate(aggregateFunction, new SOQL.QueryField(field), operator, value);
   }
 
+  global AggregateQuery havingAggregate(SOQL.Aggregate aggregateFunction, Schema.SObjectField field, SOQL.Operator operator, Object value, String bindWithKey) {
+    return this.havingAggregate(aggregateFunction, new SOQL.QueryField(field), operator, value, bindWithKey);
+  }
+
   global AggregateQuery havingAggregate(SOQL.Aggregate aggregateFunction, SOQL.QueryField queryField, SOQL.Operator operator, Object value) {
-    this.havingConditions.add(aggregateFunction.name() + '(' + queryField + ') ' + SOQL.getOperatorValue(operator) + ' ' + value);
+    return this.havingAggregate(aggregateFunction, queryField, operator, value, null);
+  }
+
```

Comment:
I think this new method overload signature makes sense, especially for trying to keep things backwards compatible with `QueryArgument`. But my small complaint is:
- I want to keep the backwards-compatability you've done for using `QueryArgument`
- I plan to always use the query with binds functionality (personal preference + my impression is that it's more performant)
- I don't want to have to specify a bind variable name every time, I'd prefer the bind variable name be auto-generated

So I'm thinking there could be another method to enable automatically generating bind variable names. Something like this:
```apex
private Boolean generateBindVariables  = false;
private Integer generatedBindVariableCounter = 0;

global AggregateQuery withBinds() {
    this.generateBindVariables  = true;
}
```

And then in this `havingAggregate()` overload, auto-generate the bind var name when enabled:
```suggestion
    String bindVariableName = this.generateBindVariables == false ? null : 'bindVar' + this.generatedBindVariableCounter++;
    return this.havingAggregate(aggregateFunction, queryField, operator, value, bindVariableName);
  }
```

Example usage would look something like this:

```apex
AggregateQuery aggregateQuery = new AggregateQuery(Schema.Account.SObjectType)
      .withBinds()
      .groupByField(Schema.Account.Name)
      .addAggregate(SOQL.Aggregate.COUNT, Schema.Account.Id)
      .havingAggregate(SOQL.Aggregate.COUNT, Schema.Account.Id, SOQL.Operator.GREATER_THAN, someVariable);
```

I think the same idea would apply in the `Query` class too. Let me know what you think.